reactivesocket-publishers/src/main/java/io/reactivesocket/reactivestreams/extensions/internal/publishers/TimeoutPublisher.java

```diff
@@ -28,6 +28,15 @@
 
 public final class TimeoutPublisher<T> implements Px<T> {
 
+    @SuppressWarnings("ThrowableInstanceNeverThrown")
+    private static final TimeoutException timeoutException = new TimeoutException();
+
+    private static final StackTraceElement[] EMPTY_STACK = new StackTraceElement[0];
```

Comment:
Rather than setting the stack trace manually, it would be better to override the `fillInStacktrace` method in all reactivesocket exceptions.

```@Override
public synchronized Throwable fillInStackTrace() {
    return this;
}
```